corehq/apps/settings/views.py

```diff
@@ -402,24 +402,25 @@ class TwoFactorSetupView(BaseMyAccountView, SetupView):
     form_list = (
         ('welcome', HQEmptyForm),
         ('method', HQTwoFactorMethodForm),
-        # other forms are dynamically added via setup_two_factor decorator
+        # other forms are registered on startup in corehq.apps.hqwebapp.apps.HqWebAppConfig
     )
 
-    @setup_two_factor
-    def setup(self, request, *args, **kwargs):
-        # this is only here to add decorators
-        return super(TwoFactorSetupView, self).setup(request, *args, **kwargs)
-
     @method_decorator(active_domains_required)
     @method_decorator(login_required)
     def dispatch(self, request, *args, **kwargs):
         # this is only here to add decorators
         return super(TwoFactorSetupView, self).dispatch(request, *args, **kwargs)
 
+    def get_form_kwargs(self, step=None):
+        kwargs = super().get_form_kwargs(step)
+        if step == 'method':
+            kwargs.setdefault('allow_phone_2fa', user_can_use_phone(self.request.couch_user))
```

Comment:
The use of `setdefault` makes it appear that there may be some scenario where _phone 2FA is allowed_ when the _user cannot use phone_. Can you explain that scenario? Otherwise, maybe this should do something like
```suggestion
            kwargs['allow_phone_2fa'] = user_can_use_phone(self.request.couch_user)
```

Reply:
Hmm good point. I was just reverting to what this was previously without much thought. Looking deeper, it looks like @mjriley initially added this with the idea of passing a 'user' object in ([here](https://github.com/dimagi/commcare-hq/commit/5f2523a80226#diff-3240f7d1a05a80cb57e5cd13140f90b74330085c1326cccbbb34733f19f9efaeR378)) but ultimately ended up passing in 'allow_phone_2fa' on [that PR](https://github.com/dimagi/commcare-hq/pull/30347/files#diff-3240f7d1a05a80cb57e5cd13140f90b74330085c1326cccbbb34733f19f9efaeR397-R400). There are other scenarios where having an existing backup phone overrides the result of `user_can_use_phone` ([here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/settings/views.py#L395-L398)), but I think in the context of this form which is specifically used when setting up 2FA for the first time or resetting it, we want to strictly prevent users who should not be using phones from doing so. I'll give Matt a chance to respond in case there is some other context I'm missing, but otherwise can make the change you've suggested to clear this up.

Reply:
It looks like you've done good digging. I don't remember how the forms worked together, but yeah, the intent was to shut off phone access unless they already had backup phones set. Daniel's approach sounds good here

Reply:
Noting that this was resolved in d94b897253b993dbba6cd400627413b6f79daf6b